src/main/java/de/tum/in/www1/artemis/service/exam/ExamSessionService.java

```diff
@@ -30,9 +30,25 @@ public class ExamSessionService {
 
     private final StudentExamRepository studentExamRepository;
 
+    private final BiFunction<Long, ExamSession, Set<ExamSession>> filterSessionsForIpAndFingerprint;
```

Comment:
Can you created these lambdas directly below where they are used? I think this would make the code more readable since you would not need to scroll up and down to find the logic.

Reply:
I defined them up there because, in my opinion, it makes no sense to define them as local variables as they are - well - not variable.